irispy/spectrograph.py

```diff
@@ -3,300 +3,156 @@
-        # Calculate measurement time of each spectrum.
-        times = [parse_time(self.meta["observation start"])+timedelta(seconds=s) for s in self.auxiliary_data["TIME"]]
-        # Convert data for each spectral window into an an
-        # xarray.DataArray and enter into data dictionary.
-        self.data = dict([(window_name, xarray.DataArray(data=data_dict[window_name],
-                                                         dims=["raster_axis", "slit_axis", "spectral_axis"],
-                                                         coords={"wavelength": ("spectral_axis",
-                                                                                spectral_coords[window_name].value),
-                                                                 "raster_position": ("raster_axis", raster_positions),
-                                                                 "time": ("raster_axis", times)},
-                                                         name="Intensity [DN]",
-                                                         attrs=OrderedDict([(
-                                                             "units", {"wavelength": spectral_coords[window_name].unit,
-                                                                       "intensity": "DN"})])))
-                          for window_name in self.spectral_windows["name"]])
-
+        # making a CubeSequence of every dictionary key window.
+        self.data = dict([(window_name, CubeSequence(data_dict[window_name], meta=self.meta, common_axis=common_axis))
+                          for window_name in self.spectral_windows['name']])
 
     def __repr__(self):
         spectral_window = self.spectral_windows["name"][0]
         spectral_windows_info = "".join(
             ["\n    {0}\n        (raster axis: {1}, slit axis: {2}, spectral axis: {3})".format(
-            name, len(self.data[name].raster_axis), len(self.data[name].slit_axis),
-            len(self.data[name].spectral_axis)) for name in self.spectral_windows["name"]])
-        return "<iris.IRISRaster instance\nOBS ID: {0}\n".format(self.meta["observation ID"]) + \
-               "OBS Description: {0}\n".format(self.meta["observation description"]) + \
-               "OBS period: {0} -- {1}\n".format(self.meta["observation start"], self.meta["observation end"]) + \
-               "Instance period: {0} -- {1}\n".format(self.data[spectral_window].time.values[0],
-                                                    self.data[spectral_window].time.values[-1]) + \
-               "Number unique raster positions: {0}\n".format(self.meta["number unique raster positions"]) + \
+                name,
+                self.data[name][0].shape[0],
+                self.data[name][0].shape[1],
+                self.data[name][0].shape[2])
+                for name in self.spectral_windows["name"]])
+        # Removed the instance Period. if required let me know
```

Comment:
Yes, please do leave the instance period.  It's a quick useful way to see how much of the OBS the instance covers.